tests/qinfo/test_fidelities.py

```diff
@@ -0,0 +1,417 @@
+        assert qml.math.allclose(fid, 1.0)
+
+    @pytest.mark.parametrize("device", devices)
+    def test_fidelity_qnodes_rx_empty(self, device):
+        """Test the fidelity between two Qnodes."""
+        dev = qml.device(device, wires=1)
+
+        @qml.qnode(dev)
+        def circuit0(x):
+            qml.RX(x, wires=0)
+            return qml.state()
+
+        @qml.qnode(dev)
+        def circuit1():
+            return qml.state()
+
+        fid = qml.qinfo.fidelity(circuit0, circuit1, wires0=[0], wires1=[0])((np.pi))
+        assert qml.math.allclose(fid, 0.0)
+
+    @pytest.mark.parametrize("device", devices)
+    def test_fidelity_qnodes_rx_empty(self, device):
+        """Test the fidelity between two Qnodes."""
+        dev = qml.device(device, wires=1)
+
+        @qml.qnode(dev)
+        def circuit0(x):
+            qml.RX(x, wires=0)
+            return qml.state()
+
+        @qml.qnode(dev)
+        def circuit1():
+            return qml.state()
+
+        fid = qml.qinfo.fidelity(circuit0, circuit1, wires0=[0], wires1=[0])((np.pi))
+        assert qml.math.allclose(fid, 0.0)
+
+    @pytest.mark.parametrize("device", devices)
+    def test_fidelity_qnodes_rx_empty(self, device):
+        """Test the fidelity between two Qnodes."""
+        dev = qml.device(device, wires=1)
+
+        @qml.qnode(dev)
+        def circuit0():
+            return qml.state()
+
+        @qml.qnode(dev)
+        def circuit1(x):
+            qml.RX(x, wires=0)
+            return qml.state()
+
+        fid = qml.qinfo.fidelity(circuit0, circuit1, wires0=[0], wires1=[0])(all_args1=(np.pi))
+        assert qml.math.allclose(fid, 0.0)
+
+    @pytest.mark.parametrize("device", devices)
+    def test_fidelity_qnodes_rxrz_rxry(self, device):
+        """Test the fidelity between two Qnodes."""
+        dev = qml.device(device, wires=1)
+
+        @qml.qnode(dev)
+        def circuit0(x, y):
+            qml.RX(x, wires=0)
+            qml.RZ(y, wires=0)
+            return qml.state()
+
+        @qml.qnode(dev)
+        def circuit1(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=0)
+            return qml.state()
+
+        fid = qml.qinfo.fidelity(circuit0, circuit1, wires0=[0], wires1=[0])(
+            (0.0, np.pi), (0.0, 0.0)
+        )
+
+        assert qml.math.allclose(fid, 1.0)
+
+    parameters = np.linspace(0, 2 * np.pi, 50)
+    wires = [1, 2]
+
+    @pytest.mark.parametrize("device", devices)
+    @pytest.mark.parametrize("param", parameters)
+    @pytest.mark.parametrize("wire", wires)
+    def test_fidelity_qnodes_rx_pauliz(self, device, param, wire):
+        """Test the fidelity between Rx and PauliZ circuits."""
+        dev = qml.device(device, wires=1)
+
+        @qml.qnode(dev)
+        def circuit0(x):
+            qml.RX(x, wires=0)
+            return qml.state()
+
+        @qml.qnode(dev)
+        def circuit1():
+            qml.PauliZ(wires=0)
```

Comment:
Applying PauliZ here leaves the state in `|0>`. Could there be a more interesting case to test (or potentially just remove this PauliZ application)?

Reply:
Indeed I added it to have something in the circuit and at the same making it too complicated to get an analytical formula.

Reply:
I see! Maybe worth a comment then, that PauliZ is just added so that there's an operation in the qfunc.